common/containers/Tabs/Swap/index.js

```diff
@@ -0,0 +1,120 @@
+import React, {Component} from 'react';
+import WantToSwapMy from './components/wantToSwapMy';
+import CurrentRates from './components/currentRates';
+import {connect} from 'react-redux';
+import {
+    SWAP_DESTINATION_AMOUNT_TO,
+    SWAP_DESTINATION_KIND_TO,
+    SWAP_ORIGIN_AMOUNT_TO,
+    SWAP_ORIGIN_KIND_TO,
+    SWAP_UPDATE_BITY_RATES_TO
+} from 'actions/swap';
+
+import PropTypes from 'prop-types';
+import Bity from 'api/bity';
+
+class Swap extends Component {
+    constructor(props) {
+        super(props);
+        this.bity = new Bity();
+    }
+
+    static propTypes = {
+        bityRates: PropTypes.any,
+        originAmount: PropTypes.any,
+        destinationAmount: PropTypes.any,
+        originKind: PropTypes.string,
+        destinationKind: PropTypes.string,
+        destinationKindOptions: PropTypes.array,
+        originKindOptions: PropTypes.array,
+        swapOriginKindTo: PropTypes.func,
+        swapDestinationKindTo: PropTypes.func,
+        swapOriginAmountTo: PropTypes.func,
+        swapDestinationAmountTo: PropTypes.func,
+        swapUpdateBityRatesTo: PropTypes.func
+
+    };
+
+    render() {
+        let {
+            bityRates,
+            originAmount,
+            destinationAmount,
+            originKind,
+            destinationKind,
+            destinationKindOptions,
+            originKindOptions,
+            swapOriginKindTo,
+            swapDestinationKindTo,
+            swapOriginAmountTo,
+            swapDestinationAmountTo
+        } = this.props;
+
+        let wantToSwapMyProps = {
+            bityRates,
+            originAmount,
+            destinationAmount,
+            originKind,
+            destinationKind,
+            destinationKindOptions,
+            originKindOptions,
+            swapOriginKindTo,
+            swapDestinationKindTo,
+            swapOriginAmountTo,
+            swapDestinationAmountTo
+        };
+
+
+        if (!bityRates.ETHBTC || !bityRates.ETHREP || !bityRates.BTCETH || !bityRates.BTCREP) {
+            this.bity.getAllRates()
```

Comment:
Probably want to move this into another section of the component (onComponentMount maybe?)

It may also be nice to add a property `loading` to the state that gets set to `true` once we make the call, and then is `false` when we hear back from them. That loading state can be used to hide / show elements, however we need.